test/runtime/fqdn.go

```diff
@@ -421,6 +440,99 @@ var _ = Describe("RuntimeFQDNPolicies", func() {
 		}
 
 	})
+
+	It("CNAME follow", func() {
+
+		By("Testing one level of CNAME")
+		policy := `
+[
+	{
+		"labels": [{
+			"key": "CNAME follow one level"
+		}],
+		"endpointSelector": {
+			"matchLabels": {
+				"container:id.app1": ""
+			}
+		},
+		"egress": [
+			{
+				"toPorts": [{
+					"ports":[{"port": "53", "protocol": "ANY"}],
+					"rules": {
+						"dns": [
+							{"matchPattern": "*.cilium.test"}
+						]
+					}
+				}]
+			},
+			{
+				"toFQDNs": [{
+					"matchName": "level1CNAME.cilium.test"
+				}]
+			}
+		]
+	}
+]`
+
+		_, err := vm.PolicyRenderAndImport(policy)
+		Expect(err).To(BeNil(), "Policy cannot be imported")
+
+		expectFQDNSareApplied()
+		target := "http://level1CNAME.cilium.test"
+		res := vm.ContainerExec(helpers.App1, helpers.CurlFail(target))
+		res.ExpectSuccess("Container %q cannot access to %q when should work", helpers.App1, target)
+
+		By("Testing three level CNAME to same target still works")
+		target = "http://level3CNAME.cilium.test"
+		res = vm.ContainerExec(helpers.App1, helpers.CurlFail(target))
+		res.ExpectSuccess("Container %q cannot access to %q when should work", helpers.App1, target)
+
+		By("Testing other CNAME in same domain should fail")
+		target = "http://world2CNAME.cilium.test"
+		res = vm.ContainerExec(helpers.App1, helpers.CurlFail(target))
+		res.ExpectFail("Container %q can access to %q when shouldn't work", helpers.App1, target)
+
+		By("Testing three level of CNAME")
+		policy = `
+[
+	{
+		"labels": [{
+			"key": "CNAME follow three levels"
+		}],
+		"endpointSelector": {
+			"matchLabels": {
+				"container:id.app2": ""
+			}
+		},
+		"egress": [
+			{
+				"toPorts": [{
+					"ports":[{"port": "53", "protocol": "ANY"}],
+					"rules": {
+						"dns": [
+							{"matchPattern": "*.cilium.test"}
+						]
+					}
+				}]
+			},
+			{
+				"toFQDNs": [{
+					"matchName": "level3CNAME.cilium.test"
+				}]
+			}
+		]
+	}
+]`
+
+		_, err = vm.PolicyRenderAndImport(policy)
+		Expect(err).To(BeNil(), "Policy cannot be imported")
+
+		expectFQDNSareApplied()
+		target = "http://level3CNAME.cilium.test"
+		res = vm.ContainerExec(helpers.App1, helpers.CurlFail(target))
+		res.ExpectSuccess("Container %q cannot access to %q when should work", helpers.App1, target)
```

Comment:
`helpers.App2`, no? for both the comment and the exec? unless you wanted the policy to apply to app1